cg/store/crud/read.py

```diff
@@ -623,12 +623,13 @@ def get_samples_by_customers_and_pattern(
         if pattern:
             filter_functions.extend([SampleFilter.BY_INTERNAL_ID_OR_NAME_SEARCH])
         filter_functions.append(SampleFilter.ORDER_BY_CREATED_AT_DESC)
-        return apply_sample_filter(
+        samples: list[Sample] = apply_sample_filter(
             samples=samples,
             customers=customers,
             search_pattern=pattern,
             filter_functions=filter_functions,
         ).all()
+        return samples, len(samples)
```

Comment:
The int we return should not be the same as the number of samples we return. The number of samples we return should always be the same as the page size (if specified) while the total should always refer to the total number of samples matching the filter parameters

Reply:
yes, fixed